inc/t_cose/t_cose_signature_sign_restart.h

```diff
@@ -0,0 +1,171 @@
+/*
+ * t_cose_signature_sign_restart.h
+ *
+ * Copyright (c) 2022, Laurence Lundblade. All rights reserved.
+ * Copyright (c) 2023, Arm Limited. All rights reserved.
+ * Created by Laurence Lundblade on 5/23/22.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ * See BSD-3-Clause license in README.md
+ */
+
+
+#ifndef t_cose_signature_sign_restart_h
+#define t_cose_signature_sign_restart_h
+
+#include "t_cose/t_cose_signature_sign.h"
+#include "t_cose/t_cose_common.h"
+#include "t_cose/t_cose_parameters.h"
+#include "t_cose/t_cose_key.h"
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+struct t_cose_signature_sign_restart {
+    /* Private data structure */
+
+    /* t_cose_signer must be the first item for the polymorphism to
+     * work.  This structure, t_cose_signature_sign_restart, will sometimes be
+     * uses as a t_cose_signature_sign.
+     */
+    struct t_cose_signature_sign s;
+
+    /* Mostly specific to ECDSA signing */
+    int32_t                      cose_algorithm_id;
+    struct t_cose_key            signing_key;
+    void                        *crypto_context;
+    uint32_t                     option_flags; // TODO: use or get rid of
+    struct t_cose_parameter      local_params[2];
+    struct t_cose_parameter     *added_signer_params;
+
+    /*Context for restartable signing. */
+    bool                  started;
+    struct q_useful_buf_c tbs_hash;
+    uint8_t               c_buffer_for_tbs_hash[T_COSE_CRYPTO_MAX_HASH_SIZE];
```

Comment:
Probably better to compute a max size based on the signing algorithms the restart signer supports. create_tbs doesn't need to know the max size. It just works off the buffer passed in.